packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,354 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { Download, Upload, IZosFilesResponse, IDownloadOptions, IUploadOptions } from "@zowe/zos-files-for-zowe-sdk";
+import { AbstractSession, IHandlerParameters, ImperativeError, ProcessUtils, GuiResult,
+    TextUtils, IDiffNameOptions, CliUtils } from "@zowe/imperative";
+import { CompareBaseHelper } from "../compare/CompareBaseHelper";
+import { existsSync, unlinkSync } from "fs";
+import { tmpdir } from "os";
+import path = require("path");
+import LocalfileDatasetHandler from "../compare/lf-ds/LocalfileDataset.handler";
+import LocalfileUssHandler from "../compare/lf-uss/LocalfileUss.handler";
+
+
+/**
+ * enum of prompts to be used as input to {@link EditUtilities.promptUser} during the file editing process
+ * @export
+ * @enum
+ */
+export enum Prompt {
+    useStash,
+    doneEditing,
+    continueToUpload,
+    viewUpdatedRemote
+}
+
+/**
+ * The possible types of file {@link ILocalFile}
+ * @export
+ * @type
+ */
+export type EditFileType = "uss" | "ds";
+
+/**
+ * A class to hold pertinent information about the local file during the editing process
+ * @export
+ * @interface
+ */
+export interface ILocalFile {
+    tempPath: string | null;
+    fileName: string;
+    fileType: EditFileType;
+    guiAvail: boolean;
+    zosResp: IZosFilesResponse | null;
+    encoding?: string | null;
+}
+
+/**
+ * A shared utility class that uss and ds handlers use for local file editing
+ * @export
+ * @class
+ */
+export class EditUtilities {
+    /**
+     * Builds a temp path where local file will be saved. If uss file, file name will be hashed
+     * to prevent any conflicts with file naming. A given filename will always result in the
+     * same unique file path.
+     * @param {ILocalFile} lfFile - object containing pertinent information about the local file during the editing process
+     * @returns {Promise<string>} - returns unique file path for temp file
+     * @memberof EditUtilities
+     */
+    public static async buildTempPath(lfFile: ILocalFile, commandParameters: IHandlerParameters): Promise<string>{
+        // find the appropriate extension for either uss or ds in two (long) operations
+        const ussExt = (lfFile.fileType === 'uss' && lfFile.fileName.includes(".")) ? lfFile.fileName.split(".").pop() : "";
```

Comment:
The --extension option gets ignored for USS files. I think ussExt should be set based on the --extension option provided by the user. If no --extension is specified, then use the extension of the file from its USS file name (how it works currently).

Reply:
> > I understand --extension to mainly be a flag for datasets not uss files
> 
> That's fair, but if `--extension` doesn't apply to the `files edit uss` command, then I think we should remove that option from this file:
> 
> https://github.com/zowe/zowe-cli/blob/987abaf2a1718dc16d0fd4494e128b98b8188038/packages/cli/src/zosfiles/edit/uss/USSFile.definition.ts#L43

Sounds good